### PR TITLE
Use new Async API for 2.0

### PR DESCRIPTION
**Pull request by blasten**

### CODE REVIEW

**Comment by frankiefu** [comment]
Lazy import doesn't work anymore.  For example, in `demo/demo2.html`, comment out `<link rel="import" href="../app-header/app-header.html">` and after the demo is loaded, add the `app-header.html` import back and you will see the header is not positioned correctly.

File: app-layout-behavior/app-layout-behavior.html [hunk 1]
```diff
@@ -48,26 +48,30 @@
      * @method resetLayout
      */
     resetLayout: function() {
-      this._debounceRaf(this._updateLayoutStates);
-      this._notifyDescendantResize();
```
Comment (frankiefu) [inline app-layout-behavior/app-layout-behavior.html]: Why `_notifyDescendantResize` is removed?

File: app-layout-behavior/app-layout-behavior.html [hunk 2]
```diff
@@ -48,26 +48,30 @@
...
-    _debounceRaf: function(fn) {
-      var self = this;
-      if (this._raf) {
-        window.cancelAnimationFrame(this._raf);
-      }
-      this._raf = window.requestAnimationFrame(function() {
-        self._raf = null;
-        fn.call(self);
-      });
+      this.fire('app-reset-layout');
```
Comment (frankiefu) [inline app-layout-behavior/app-layout-behavior.html]: I think we still need this inside requestAnimationFrame?

File: app-header-layout/test/app-header-layout.html [hunk 3]
```diff
@@ -127,23 +116,18 @@
...
+        flush(function() {
+          var header = document.createElement('app-header');
+          header.innerHTML = '<app-toolbar></app-toolbar>'
+          header.slot = 'header';
+          header.fixed = true;
+          Polymer.dom(headerLayout).appendChild(header);
+          Polymer.dom.flush();
+          assert.equal(headerLayout.$.contentContainer.style.marginTop,
+              header.offsetHeight + 'px');
+           });
```
Comment (frankiefu) [inline app-header-layout/test/app-header-layout.html]: This changes technically not related to this PR but instead is to fixup the failing tests from the [previous PR](https://github.com/PolymerElements/app-layout/pull/416).  Do you mind doing a separate PR so we can get this merge sooner?  So we can have the tests passing in travis again.  Thanks!

Comment (blasten) [inline app-header-layout/test/app-header-layout.html]: `Polymer.dom.flush();` fixes the timing problem, but I can only use it now with the new async API.

File: app-layout-behavior/app-layout-behavior.html (hunk 1 above)
Comment (blasten) [inline app-layout-behavior/app-layout-behavior.html]: added back. sorry

File: app-layout-behavior/app-layout-behavior.html (hunk 2 above)
Comment (blasten) [inline app-layout-behavior/app-layout-behavior.html]: I put it back although I think we are missing something like Polymer.Async.afterPaint: `requestAnimationFrame(() => setTimeout(cb));` That way, the rAF i'm adding back can be removed and the issue is fixed I think.

**Comment by frankiefu** [comment]
LGTM

**Pull request merged**